Maple2.Server.Game/Manager/ItemBoxManager.cs

```diff
@@ -111,7 +96,9 @@ public class ItemBoxManager {
 
         // Get common dropbox
         if (!int.TryParse(parameters["commonBoxId"], out int commonBoxId) ||
-            !session.TableMetadata.IndividualItemDropTable.Entries.TryGetValue(commonBoxId, out Dictionary<byte, IList<IndividualItemDropTable.Entry>>? commonDropGroupTable)) {
+            // we're doing a check preemptively here for safety in order to not consume keyItem if the box doesn't exist
+            !session.ServerTableMetadata.IndividualDropItemTable.Entries.TryGetValue(commonBoxId, out IDictionary<int, IndividualDropItemTable.Entry>? entryDict) ||
+            entryDict.Count == 0) {
```

Comment:
i think this should go in a variable above the if and all the same occasions below.

maybe a helper function somewhere

Reply:
This is only utilized once in ItemBoxManager (one per different box type). I'm not sure what the best way to go about this would be.

Reply:
then just a variable i think its enough